exporter/datadogexporter/config.go

```diff
@@ -0,0 +1,145 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datadogexporter
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/config/confignet"
+)
+
+var (
+	errUnsetAPIKey = errors.New("the Datadog API key is unset")
```

Comment:
Nit: it may be more user friendly to mention the actual config setting that is missing:
```suggestion
	errUnsetAPIKey = errors.New("api.key is not set")
```